apps/avatax/src/modules/saleor/shipping-line.ts

```diff
@@ -1,39 +1,90 @@
-import { LineItemModel } from "avatax/lib/models/LineItemModel";
+import { Result, err, ok } from "neverthrow";
+import { z } from "zod";
+import { BaseError } from "../../error";
 import { SHIPPING_ITEM_CODE } from "../avatax/calculate-taxes/avatax-shipping-line";
 
-export class SaleorShippingLine {
-  private getAmount({
-    gross,
-    net,
-    taxIncluded,
-  }: {
-    gross: number;
-    net: number;
-    taxIncluded: boolean;
-  }) {
-    return taxIncluded ? gross : net;
+export interface ISaleorOrderConfirmedShippingLine {
+  getAmount(): number;
+  getIsTaxIncluded(): boolean;
+  getIsDiscounted(): boolean;
+  getQuantity(): number;
+  getItemCode(): string;
+  getShippingTaxCode(): string | undefined;
+}
+
+export class SaleorOrderConfirmedShippingLine implements ISaleorOrderConfirmedShippingLine {
+  public static schema = z.object({
+    gross: z.object({
+      amount: z.number(),
+    }),
+    net: z.object({
+      amount: z.number(),
+    }),
+  });
+
+  private constructor(
+    private data: z.infer<typeof SaleorOrderConfirmedShippingLine.schema>,
+    // order config
+    private isTaxIncluded: boolean,
+    private isDiscounted: boolean,
+    // avatax config
+    private shippingTaxCode: string | undefined,
+  ) {}
+
+  public static ParsingError = BaseError.subclass("SaleorOrderConfirmedShippingLineParsingError");
+
+  public static create(
+    payload: unknown,
```

Comment:
Hmm, are we sure we want to have unknown? Of course once you parse TOTALLY UNKNOWN things like CSV file, that makes sense. But I think we pretty much know what we parse.